### PR TITLE
nodejs: Fix removing file that doesn't exist

### PR DESCRIPTION
**Pull request by mpozniak95**
## Related Issue
https://github.com/Granulate/gprofiler/issues/537

### CODE REVIEW

**Comment by mpozniak95** [comment]
Flaky: https://github.com/Granulate/gprofiler/issues/584

File: gprofiler/profilers/node.py [hunk 1]
```diff
@@ -223,9 +223,10 @@ def get_node_processes() -> List[psutil.Process]:
     return pgrep_exe(r".*node[^/]*$")
 
 
-def generate_map_for_node_processes(processes: List[psutil.Process]) -> None:
+def generate_map_for_node_processes(processes: List[psutil.Process]) -> List[int]:
     """Iterates over all NodeJS processes, starts debugger for it, finds debugger URL,
     copies node-linux-perf module into process' namespace, loads module and starts it."""
+    pids_with_perf_map = []
```
Comment (Jongy) [inline gprofiler/profilers/node.py:229]: Keep `Process` objects, not PIDs. PIDs may be reused but a `Process` object can always be tested against the same process and you cannot possibly refer a different one

File: gprofiler/profilers/perf.py [hunk 2]
```diff
@@ -183,6 +183,7 @@ def __init__(
         self._metadata_collectors: List[PerfMetadata] = [GolangPerfMetadata(stop_event), NodePerfMetadata(stop_event)]
         self._insert_dso_name = insert_dso_name
         self._node_processes: List[Process] = []
+        self.pids_with_perf_maps: List[int] = []
```
Comment (Jongy) [inline gprofiler/profilers/perf.py:186]:
```suggestion
        self._node_processes_attached: List[int] = []
```

`_` to signify it's private
and the name to match the `_node_processes` attribute

File: gprofiler/profilers/perf.py [hunk 3]
```diff
@@ -218,14 +219,14 @@ def start(self) -> None:
...
-            generate_map_for_node_processes(self._node_processes)
+            self.pids_with_perf_maps.extend(generate_map_for_node_processes(self._node_processes))
         for perf in self._perfs:
             perf.start()
 
     def stop(self) -> None:
         if self.perf_node_attach:
             self._node_processes = [process for process in self._node_processes if is_process_running(process)]
-            clean_up_node_maps(self._node_processes)
+            clean_up_node_maps(self._node_processes, self.pids_with_perf_maps)
```
Comment (Jongy) [inline gprofiler/profilers/perf.py:229]: You can call this function on `self._node_processes_attached` instead

**Review by Jongy — CHANGES_REQUESTED** [review]
Done reviewing

**Review by Jongy — APPROVED** [review]

**Pull request merged**